cmd/pctl/add.go

```diff
@@ -245,6 +250,25 @@ func getGitRepositoryNamespaceAndName(c *cli.Context) (string, string, error) {
 	return "", "", fmt.Errorf("flux git repository not provided, please provide the --git-repository flag or use the pctl bootstrap functionality")
 }
 
+// getOutFolder returns the output folder with the following precedence:
+// User set --out overrides local configuration.
+// Local configuration, if set.
+// Default out which is `.`.
+func getOutFolder(c *cli.Context) (string, error) {
+	if c.IsSet("out") {
+		return c.String("out"), nil
+	}
+	wd, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("failed to fetch current working directory: %w", err)
+	}
+	config, err := bootstrap.GetConfig(wd)
```

Comment:
I think we do this already somewhere for getting the `gitRepository` value? might be able to pass it in rather than re-fetching it

Reply:
We do, yeah. I was wondering about that, but in the end, I wanted to keep its integrity to itself? Like a single, self-sustained thing? But I'll do the refactor and see how it feels. :)

Reply:
I know now why it's separate. Because if it's set, we bail out early, not even trying to attempt to do an io operation. It's a small optimisation basically. Not saying that it's worth it, but that was why I didn't extract it. Because we have a path in which it's not used at all.

Reply:
:thinking: you mean if `out` and `git-repository` are set we don't need to read this, so the logic is separate for each call?

Going along with our chat in slack maybe always reading it, and then logging:
1. if we read it/find it
2. if we fail (for any reason)
3. what values we ignore/use

Reply:
No, in both cases if the user set something, we never read the config file. That's what I meant.

Reply:
I mean, I can still do the refactor of course. Just wanted to share why I didn't do it. :D Because there is a small chance that we never HAVE to read the file. :)

Reply:
Done